modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/handlers/CommandHandlers.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.partition.replicator.raft.handlers;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.ignite.internal.raft.WriteCommand;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * This class represents a collection of command handlers.
+ *
+ * @see AbstractCommandHandler
+ */
+public class CommandHandlers {
+    private final Map<MessageId, AbstractCommandHandler<? extends WriteCommand>> handlers = new HashMap<>();
```

Comment:
Are you sure that `HashMap` is enough? For example, `ParititionListener` is created in one thread, while commands are executed in another. I think the correct way is to make this map immutable and initialize it in the constructor.

Reply:
I think there is no issue with safe publication here. However, as we discussed in private, it would be nice to add a `Builder` class to avoid any issues.